src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -545,6 +545,8 @@ protected override int MakeSlot(BoundExpression node)
                             case ConversionKind.ExplicitTupleLiteral:
                             case ConversionKind.ImplicitTuple:
                             case ConversionKind.ExplicitTuple:
+                            case ConversionKind.Boxing:
```

Comment:
>Boxing [](start = 48, length = 6)

from our discussion, it would be good to leave a comment saying why we don't need to create new slots for boxing/unboxing (ie. assignment will do that anyways).
Also, the scenario you wrote on the whiteboard (`(S)(object)new S() { F = null }`) is a good illustration of current behavior, if that's not already tested.